### PR TITLE
Add comprehensive URL builder for OpenDAP endpoints

### PR DESCRIPTION
**Pull request by mpiannucci**
## Summary
- Implements a builder pattern for constructing OpenDAP URLs with support for DAS, DDS, and DODS endpoints
- Adds full constraint expression support for subsetting data following OpenDAP DAP2 standards
- Provides a foundational building block for WASM and higher-level implementations

## Features
- **Builder Pattern**: Fluent API for constructing OpenDAP URLs
- **Multiple Endpoints**: Support for DAS, DDS, and DODS URLs  
- **Constraint Expressions**: Full support for OpenDAP constraint syntax including:
  - Single index selection: `variable[5]`
  - Range selection: `variable[0:10]`
  - Range with stride: `variable[0:2:20]`
  - Multi-dimensional constraints: `variable[0:10][5][20:50]`
  - Negative indices for longitude: `variable[-180:180]`

## API Design
- `UrlBuilder::new(base_url)` - Create a new builder
- `add_variable(name)` - Add variables to select
- `add_single_index(var, index)` - Add single index constraint
- `add_range(var, start, end, stride)` - Add range constraint
- `add_multidimensional_constraint(var, indices)` - Add complex multi-dimensional constraints
- `das_url()`, `dds_url()`, `dods_url()` - Generate endpoint URLs
- `clear_variables()`, `clear_constraints()`, `clear_all()` - Reset builder state

## Testing
- **33 unit tests** covering all basic functionality
- **8 integration tests** with real-world OpenDAP scenarios
- **100% test coverage** for constraint expression generation
- **Edge case handling** for empty constraints, zero indices, negative values

🤖 Generated with [opencode](https://opencode.ai)

### CODE REVIEW

File: readap/src/query.rs [hunk 1]
```diff
@@ -0,0 +1,665 @@
...
+        if !self.selected_variables.contains(&name.to_string()) {
+            self.selected_variables.push(name.to_string());
+        }
+
+        Ok(self)
+    }
+
+    /// Select multiple variables with validation
+    pub fn select_variables(mut self, names: &[&str]) -> Result<Self, QueryError> {
+        for name in names {
```
Comment (mpiannucci) [inline readap/src/query.rs:242]: Just call `self.select_variable` in the loop instead o repeating the logic

File: readap/src/query.rs [hunk 2]
```diff
@@ -0,0 +1,665 @@
...
+    /// Index-based range constraint with optional stride
+    Indices {
+        start: usize,
+        end: usize,
+        stride: Option<usize>,
+    },
+    /// Single index constraint
+    Single(usize),
+    /// Value-based range constraint (for future enhancement)
+    Values {
```
Comment (mpiannucci) [inline readap/src/query.rs]: Lets get rid of all logic in this file that concerns value based indexing, we will let higher level implementations implement this and treat the Coordinate constraint as a pure building block
